src/RestSharp/Serializers/Xml/XmlDeserializer.cs

```diff
@@ -188,7 +188,14 @@ protected virtual object Map(object x, XElement root)
                 }
                 else if (type.IsPrimitive)
                 {
-                    prop.SetValue(x, value.ChangeType(asType), null);
+                    try
+                    {
+                        prop.SetValue(x, value.ChangeType(asType), null);
+                    }
+                    catch (FormatException)
+                    {
+                        throw new FormatException("Format exception while Deserializing Property Name - " + prop.Name);
```

Comment:
Could you include the original FormatException as InnerException of the new one?

Also could you add a little more detail to the message itself. I would imagine something like this:
$"Couldn't parse the value of '{value}' into the '{prop.Name}' property, because it isn't a type of '{prop.Type}'."

Reply:
Thanks for the inputs, I have taken care of these changes. Can you please review the same?